src/ProjectTemplates/Shared/Project.cs

```diff
@@ -288,6 +289,63 @@ public void AssertFileExists(string path, bool shouldExist)
         }
     }
 
+    public async Task<Project> VerifyLaunchSettings(string[] expectedLaunchProfileNames)
+    {
+        var launchSettingsFiles = Directory.EnumerateFiles(TemplateOutputDir, "launchSettings.json", SearchOption.AllDirectories);
```

Comment:
Seems this assumes a well-ordered list of files, but I doubt that's reliable on all platforms. Suggest finding all profile names, sticking them in a sorted list, then asserting that list and `expectedLaunchProfileNames` contain the same values. This may also reduce the complexity of this method.

Reply:
Not following how this relies on the order of the files themselves.

Reply:
This enumerates files because some templates (e.g. Blazor WASM Hosted) create multiple projects with each project having their own `launchSettings.json` file. This test doesn't rely on file system ordering though, the order is coming from the JSON array inside the file itself.